packages/rocketchat-lib/server/models/Messages.js

```diff
@@ -500,6 +500,20 @@ RocketChat.models.Messages = new class extends RocketChat.models._Base {
 		return this.update(query, update);
 	}
 
+	unlinkUserId(userId, newUserId, newUsername) {
```

Comment:
Can a flag be set on the message that this message has changed owner/sender? That way future confusion can be cleared up. Obviously the original creator information can't be stored, but a flag saying this didn't originally come from the rocket.cat user and instead came from someone else.